@kiva/kv-shop/src/basketCredits.ts

```diff
@@ -51,9 +51,16 @@ export interface ApplyPromoCreditData {
 	} | null,
 }
 
-export async function applyPromoCredit(apollo: ApolloClient<any>): Promise<boolean> {
+export async function applyPromoCredit(
+	apollo: ApolloClient<any>,
+	options: MutationOptions<any>,
+): Promise<boolean> {
+	if (!options?.variables?.creditType && !options?.variables?.redemptionCode) {
+		return Promise.resolve(false);
+	}
 	const data = await callShopMutation<ApplyPromoCreditData>(apollo, {
 		awaitRefetchQueries: true,
+		fetchPolicy: options?.fetchPolicy ?? 'network-only',
```

Comment:
Do we want the default to be `network-only`? The typical default is `cache-first`.

Reply:
These should always be network only as they need to update the basket state on the server